core/sync/chan/chan.odin

```diff
@@ -164,12 +164,17 @@ send_raw :: proc "contextless" (c: ^Raw_Chan, msg_in: rawptr) -> (ok: bool) {
 	}
 	if c.queue != nil { // buffered
 		sync.guard(&c.mutex)
-		for c.queue.len == c.queue.cap {
+		for !sync.atomic_load(&c.closed) &&
```

Comment:
On second look, we might be able to change all `c.closed` loads to non-atomic, not just this one. I was following convention on this change, since the other accesses are atomic.

Reply:
Yeah, this should save an extra `lwsync` operation on ARM, iirc

Reply:
In the end, I removed all atomic loads and stores, as well as the other two mutexes. Everything is under guard by `mutex` already.